apps/client/src/app/core/data-reporting/airship-exploration-result-reporter.ts

```diff
@@ -37,23 +38,20 @@ export class AirshipExplorationResultReporter extends ExplorationResultReporter
       toIpcData()
     );
 
-    const updateHullCondition$ = packets$.pipe(
-      ofMessageType('updateInventorySlot'),
-      toIpcData(),
+    return resultLog$.pipe(
       withLatestFrom(isAirshipMenuOpen$),
-      filter(([updateInventory, isOpen]) => {
-        return isOpen && updateInventory.containerId === 25003 && [30, 35, 40, 45].includes(updateInventory.slot) && updateInventory.condition < 30000;
-      })
-    );
-
-    return updateHullCondition$.pipe(
-      withLatestFrom(status$),
-      map(([, status]) => {
-        return this.getBuildStats(status.hull, status.rigging, status.forecastle, status.aftcastle);
-      }),
-      switchMap(stats => {
-        return resultLog$.pipe(
-          map(resultLog => this.createReportsList(stats, resultLog))
+      filter(([, isOpen]) => isOpen),
+      switchMap(([resultLog]) => {
+        return status$.pipe(
+          filter((status: AirshipStatus) => {
+            return this.shouldSendReport(status.returnTime, resultLog.airshipSpeed, resultLog.explorationResult);
```

Comment:
Aaah.... How fun, the `AirshipStatus` doesn't have the `birthDate`. It might be better to do like submarines then. Use the `UpdateInventorySlot` to figure out the ship slot then get its `birthDate` from the `AirshipStatusList`

Reply:
I think the speed is enough, because returnTime + result is already almost good enough for Unicity imo.